### DISCUSSION TITLE
Setting is_keyframe on Packet

### DISCUSSION THREAD

**Discussion by daveisfera** [PyAV-Org/PyAV#1179] [1. Help]
I have encoded video data that I'm adding to a `Packet` that I create, but then I can't set `is_keyframe` before passing the packet to `mux` of the container. Is there a way to do that or does the code need to be changed to allow that?

(I assume that would be [adding `__set__` to the `is_keyframe` property](https://github.com/PyAV-Org/PyAV/blob/6982d818080716b39bc3f333cf19daa0a3e0b56a/av/packet.pyx#L196) and I'm willing to submit a PR for that, if that's the needed fix)

**Comment by JoeSchiff** [comment]
@daveisfera Were you able to set keyframes (or `packet.is_corrupt`) after merging your PR https://github.com/PyAV-Org/PyAV/pull/1208? I tried the following code, but it doesn't seem to have any effect:
```
import av
from av import VideoFrame
from fractions import Fraction

PATH = "\home\joepers\temp.mp4"
WIDTH = 320
HEIGHT = 240
DURATION = 5

with av.open(PATH, "w") as output:
    stream = output.add_stream("mpeg4", 24)
    stream.width = WIDTH
    stream.height = HEIGHT
    stream.pix_fmt = "yuv420p"
    for i in range(DURATION):
        frame = VideoFrame(WIDTH, HEIGHT, "rgb24")
        frame.pts = i * 2000
        frame.time_base = Fraction(1, 48000)
        for packet in stream.encode(frame):
            packet.is_keyframe = False
            output.mux(packet)
            print(f"Encoding: {packet.is_keyframe=}")

with av.open(PATH, "r") as container:
    stream = container.streams.video[0]
    for packet in container.demux(stream):
        for frame in packet.decode():
            print(f"Decoding: {packet.is_keyframe=}")
```
Output:
```
Encoding: packet.is_keyframe=False
Encoding: packet.is_keyframe=False
Encoding: packet.is_keyframe=False
Encoding: packet.is_keyframe=False
Encoding: packet.is_keyframe=False
Decoding: packet.is_keyframe=True
Decoding: packet.is_keyframe=True
Decoding: packet.is_keyframe=True
Decoding: packet.is_keyframe=True
Decoding: packet.is_keyframe=True
```

**Reply by daveisfera** [reply]
I believe that `is_keyframe` defaults to `False` so I think that something else is going on with your test

**Comment by jimibit** [comment]
@daveisfera 

Hi, try to set the gop_size: 

```
with av.open(PATH, "w") as output:
    stream = output.add_stream("h264", rate=24)
    stream.width = WIDTH
    stream.height = HEIGHT
    stream.pix_fmt = "yuv420p"
    stream.gop_size = 16

```
in this way, when you mux, the key_frames will be 1 every 16.

Bye

**Reply by daveisfera** [reply]
I submitted a PR to allow setting the value:
https://github.com/PyAV-Org/PyAV/pull/1208
